math/minuit2/src/MnHesse.cxx

```diff
@@ -147,17 +164,65 @@ MinimumState MnHesse::ComputeAnalytical(const FCNBase & fcn, const MinimumState
    return MinimumState(st.Parameters(), err, gr, edm, st.NFcn());
 }
 
+namespace {
 
-MinimumState MnHesse::ComputeNumerical(const MnFcn &mfcn, const MinimumState &st, const MnUserTransformation &trafo,
-                                 unsigned int maxcalls) const
+// Helper class to call the MnFcn, cashing the transformed parameters in case
+// it is a MnUserFcn that does the parameter transformation when calling.
+class MnFcnCaller {
+public:
+   MnFcnCaller(const MnFcn &mfcn) : _mfcn{mfcn}, _isUserFcn{static_cast<bool>(dynamic_cast<MnUserFcn const *>(&mfcn))}
+   {
+      if (!_isUserFcn)
+         return;
+
+      MnUserTransformation const &transform = static_cast<MnUserFcn const &>(_mfcn).transform();
+
+      // get first initial values of parameter (in case some one is fixed)
+      _vpar.assign(transform.InitialParValues().begin(), transform.InitialParValues().end());
+   }
+
+   double operator()(const MnAlgebraicVector &v)
+   {
+      if (!_isUserFcn)
+         return _mfcn(v);
+
+      MnUserTransformation const &transform = static_cast<MnUserFcn const &>(_mfcn).transform();
+
+      bool firstCall = _lastInput.size() != v.size();
+
+      _lastInput.resize(v.size());
+
+      for (unsigned int i = 0; i < v.size(); i++) {
+         if (firstCall || _lastInput[i] != v(i)) {
+            _vpar[transform.ExtOfInt(i)] = transform.Int2ext(i, v(i));
+            _lastInput[i] = v(i);
+         }
+      }
+
+      return static_cast<MnUserFcn const &>(_mfcn).callWithTransformedParams(_vpar);
+   }
+
+private:
+   MnFcn const &_mfcn;
```

Comment:
To be consistent, in Minuit we use the ROOT coding convention, data members should start with f + capital letters instead of "_". 
Ex:
_mfcn; -> fMfcn

Reply:
Ah yes, I was thinking about RooFit. No problem, I'll update that